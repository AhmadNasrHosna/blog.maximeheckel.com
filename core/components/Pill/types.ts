export interface PillProps extends React.HTMLAttributes<HTMLSpanElement> {
  children: React.ReactNode;
  variant: 'info' | 'danger' | 'success' | 'warning';
}
