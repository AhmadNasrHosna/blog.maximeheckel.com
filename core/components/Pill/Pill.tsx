import { useTheme } from '@theme/context/ThemeContext';
import { PillProps } from './types';
import { StyledPill } from './Styles';

const Pill = (props: PillProps) => {
  const theme = useTheme();
  const { children, variant } = props;
  return (
    <StyledPill {...props} dark={theme.dark} variant={variant}>
      {children}
    </StyledPill>
  );
};

export default Pill;
