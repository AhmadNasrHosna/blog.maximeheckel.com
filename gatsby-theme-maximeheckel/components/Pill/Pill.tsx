import React from 'react';
import styled from '@emotion/styled';

interface PillWrapperProps {
  color: string;
}

const PillWrapper = styled('span')<PillWrapperProps>`
  font-size: 14px;
  background: ${(p) => p.color};
  border-radius: var(--border-radius-1);
  height: 28px;
  display: inline-flex !important;
  align-items: center;
  justify-content: center;
  padding: 5px 8px 5px 8px !important;
  color: var(--maximeheckel-colors-brand);
  border: 1px solid var(--maximeheckel-colors-brand);
  min-width: 40px;
  font-weight: 500;
  cursor: default;
  user-select: none;
`;

interface Props extends PillWrapperProps {
  text: string;
}

const Pill: React.FC<Props> = (props) => {
  const { text, color } = props;
  return (
    <PillWrapper {...props} color={color}>
      {text}
    </PillWrapper>
  );
};

export { Pill };
