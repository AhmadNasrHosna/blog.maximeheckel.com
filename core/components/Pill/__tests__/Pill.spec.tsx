import { cleanup, render } from '@testing-library/react';
import React from 'react';
import Pill from '../';

afterEach(() => {
  cleanup();
});

describe('Pill', () => {
  it('Renders the Pill component with the info variant', () => {
    const { getByText, asFragment } = render(<Pill variant="info">test</Pill>);
    expect(getByText('test')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });

  it('Renders the Pill component with the success variant', () => {
    const { getByText, asFragment } = render(
      <Pill variant="success">test</Pill>
    );
    expect(getByText('test')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });

  it('Renders the Pill component with the danger variant', () => {
    const { getByText, asFragment } = render(
      <Pill variant="danger">test</Pill>
    );
    expect(getByText('test')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });

  it('Renders the Pill component with the warning variant', () => {
    const { getByText, asFragment } = render(
      <Pill variant="warning">test</Pill>
    );
    expect(getByText('test')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });
});
