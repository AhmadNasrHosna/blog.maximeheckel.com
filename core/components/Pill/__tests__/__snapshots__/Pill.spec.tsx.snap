// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pill Renders the Pill component with the danger variant 1`] = `
<DocumentFragment>
  <span
    class="c-gIiHtB c-gIiHtB-cGyTej-variant-danger"
  >
    test
  </span>
</DocumentFragment>
`;

exports[`Pill Renders the Pill component with the info variant 1`] = `
<DocumentFragment>
  <span
    class="c-gIiHtB c-gIiHtB-dGJyPl-variant-info"
  >
    test
  </span>
</DocumentFragment>
`;

exports[`Pill Renders the Pill component with the success variant 1`] = `
<DocumentFragment>
  <span
    class="c-gIiHtB c-gIiHtB-ypDbH-variant-success"
  >
    test
  </span>
</DocumentFragment>
`;

exports[`Pill Renders the Pill component with the warning variant 1`] = `
<DocumentFragment>
  <span
    class="c-gIiHtB c-gIiHtB-lghveN-variant-warning"
  >
    test
  </span>
</DocumentFragment>
`;
